R/workflow.R

```diff
@@ -71,13 +82,24 @@ workflow <- R6::R6Class(
         error = function(e) {
           orderly_log("error", sprintf("Running report '%s' failed", report))
           stop(e)
-        })
+        },
+        finally = git_restore())
       }
       orderly_log("workflow",
                   sprintf("Completed running workflow '%s' with ID '%s'",
                           self$workflow_name, self$workflow_id))
       run_ids
     }
+  ),
+
+  private = list(
+    git_checkout = function(ref) {
```

Comment:
Just looking at this makes me think we really need to start thinking about https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-3900 at some point...